sdk/storage/Azure.Storage.DataMovement/src/Shared/TransferManager.cs

```diff
@@ -260,6 +261,32 @@ public virtual Task<bool> TryPauseTransferAsync(string transferId, CancellationT
             }
         }
 
+        /// <summary>
+        /// Gets the current transfers stored in the <see cref="TransferManager"/>.
+        /// </summary>
+        /// <param name="filterByStatus">
+        /// If specified, the returned list of transfers will have only have the transfers
+        /// of which match the status specified.
+        ///
+        /// If not specified or specified to <see cref="StorageTransferStatus.None"/>,
+        /// all transfers will be returned regardless of status.
+        /// </param>
+        /// <returns></returns>
+        public virtual List<DataTransfer> GetTransfers(
```

Comment:
Two comments:
- Do we store completed/failed transfers? If so, for how long? If we do not and this is only active transfers, maybe it makes sense to rename this API to something like `GetActiveTransfers` or `GetCurrentTransfers`?
- Should we return `IEnumerable` rather than enumerating (with `ToList`) for the user? Kind of a nit and if we do not usually return `IEnumerable` in other places, probably fine to leave out.

Reply:
We do store the completed / failed transfers. There's actually not a time limit on when we evict the transfer from the transfers stored. We actually would return all transfers of all states.

I'm okay with returning an IEnumerable as well

Reply:
If `TransferManager` ever becomes singleton we may want to consider if we want to remove transfers but for now that's probably fine. In this case, API name seems fine to me. I could also see an argument for something like `GetAllTransfers` but ehh.

Reply:
Updated to IEnumerable, but `GetAllTransfers` won't be a thing since, there's a filter option by status.

Also just following our `GetBlobs` API as a precedence, we don't have `GetAllBlobs` it's just `GetBlobs` with filters and options.